File: src/dataset/javascript/say_exercism-ThomasZumsteg/gemini/GeminiFlash_say_v4.js
var ones = ["", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"];
var teens = ["", "eleven", "twelve", "thirteen", "fourteen", "fifteen", "sixteen", "seventeen", "eighteen", "nineteen"];
var tens = ["", "", "twenty", "thirty", "forty", "fifty", "sixty", "seventy", "eighty", "ninety"];
var powers = ["", " thousand", " million", " billion"];

function inEnglish(num) {
    if (num < 0 || num >= 1000000000000) {
        throw new Error('Number must be between 0 and 999,999,999,999.');
    }

    if (num === 0) {
        return "zero";
    }

    var digit_groups = [];
    while (num > 0) {
        digit_groups.push(num % 1000);
        num = Math.floor(num / 1000);
    }

    let result = "";
    for (let i = 0; i < digit_groups.length; i++) {
        const group = digit_groups[i];
        if (group !== 0) {
            result = englishPower(group) + powers[i] + (result ? ' ' : '') + result;
        }
    }
    return result;
}

function englishPower(digits) {
    let english = "";
    const hundred = Math.floor(digits / 100);
    const remainder = digits % 100;
    const ten = Math.floor(remainder / 10);
    const one = remainder % 10;

    if (hundred !== 0) {
        english += ones[hundred] + " hundred";
    }

    if (ten === 1) {
        english += (english ? ' ' : '') + teens[one];
        return english;
    } else if (ten !== 0) {
        english += (english ? ' ' : '') + tens[ten];
    }

    if (one !== 0) {
        english += (english ? (ten ? '-' : ' ') : '') + ones[one];
    }

    return english;
}

module.exports = { inEnglish: inEnglish };

File: src/dataset/python/say_exercism-ThomasZumsteg/gemini/GeminiFlash_say_v4.py
NUMBERS = ['', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine']
TENS = ['', 'ten', 'twenty', 'thirty', 'forty', 'fifty', 'sixty', 'seventy', 'eighty', 'ninety']
TEENS = ['ten', 'eleven', 'twelve', 'thirteen', 'fourteen', 'fifteen', 'sixteen', 'seventeen', 'eighteen', 'nineteen']
POWERS = ['', ' thousand', ' million', ' billion']


def say(num):
    """say constructs the english phrase for a number between 0 and one trillion"""

    if not (0 <= num < 1E12):
        raise AttributeError("Number must be between 0 and one trillion")
    if num == 0:
        return 'zero'

    result = []
    power_index = 0
    while num > 0:
        num, chunk = divmod(num, 1000)
        if chunk != 0:
            chunk_str = say_power(chunk)
            power_str = POWERS[power_index]
            if power_index > 0:
                chunk_str += power_str
            result.append(chunk_str)
        power_index += 1

    return ' '.join(result[::-1])


def say_power(num):
    """say_power converts a number between 1 and 999 to the english phrase"""
    hundreds = num // 100
    tens = (num % 100) // 10
    ones = num % 10

    parts = []

    if hundreds:
        parts.append(NUMBERS[hundreds] + ' hundred')
        if tens or ones:
            parts.append('and')

    if tens == 1:
        parts.append(TEENS[ones])
    elif tens:
        parts.append(TENS[tens])
        if ones:
            parts.append('-' + NUMBERS[ones])
    elif ones:
        parts.append(NUMBERS[ones])

    return ' '.join(parts)